framework/xds_k8s_testcase.py

```diff
@@ -691,6 +691,84 @@ def assertAllBackendsReceivedRpcs(self, lb_stats):
                 msg=f"Backend {backend} did not receive a single RPC",
             )
 
+    def assertClientEventuallyReachesSteadyState(
+        self,
+        test_client: XdsTestClient,
+        rpc_type: str,
+        num_rpcs: int,
+        threshold_percent: int,
+        *,
+        retry_timeout: dt.timedelta = dt.timedelta(minutes=20),
+        retry_wait: dt.timedelta = dt.timedelta(seconds=2),
+        steady_state_delay: dt.timedelta = dt.timedelta(seconds=5),
+    ):
+        retryer = retryers.constant_retryer(
+            wait_fixed=retry_wait,
+            timeout=retry_timeout,
+            error_note=(
+                f"Timeout waiting for test client ${test_client.hostname} to"
+                f"report {num_rpcs} pending calls +/-{threshold_percent}%"
+            ),
+        )
+        for attempt in retryer:
+            with attempt:
+                self._checkRpcsInFlight(
+                    test_client, rpc_type, num_rpcs, threshold_percent
+                )
+        time.sleep(steady_state_delay.total_seconds())
```

Comment:
Important - we need a loger.info message here before sleeping. It should contain:
1. why we sleep 
2. for how long

It confused me in the test log why we have seemingly successful check ran twice:

```md
I0507 21:53:59.044477 129579095617536 xds_k8s_testcase.py:745] [psm-grpc-client-6f65d8fcf5-b67bn] << EMPTY_CALL RPCs in flight: 1000, expecting 1000 +/-1%
...
I0507 21:54:04.053826 129579095617536 xds_k8s_testcase.py:745] [psm-grpc-client-6f65d8fcf5-b67bn] << EMPTY_CALL RPCs in flight: 1000, expecting 1000 +/-1%
```

I thought something is wrong with the check function first. We need to be clear that this is an extra step we take in the log, otherwise the triager will have no idea what's happening.